### PR TITLE
Remove unuseful null check on Activator.CreateInstance

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
During the developing/testing of new Activator.CreateInstance apis we found unuseful null check.

Comment https://github.com/dotnet/corefx/pull/30809#discussion_r200356834

/cc @jkotas

### CODE REVIEW

File: src/System.Private.CoreLib/src/System/Activator.cs [hunk 1]
```diff
@@ -164,7 +164,7 @@ public static ObjectHandle CreateInstance(string assemblyName, string typeName,
                                                 culture,
                                                 activationAttributes);
 
-            return (o != null) ? new ObjectHandle(o) : null;
```
Comment (jkotas) [inline src/System.Private.CoreLib/src/System/Activator.cs]: You should be able to hit this path with null for `Nullable<int>`

File: src/System.Private.CoreLib/src/System/Activator.cs [hunk 2]
```diff
@@ -164,7 +164,7 @@ public static ObjectHandle CreateInstance(string assemblyName, string typeName,
                                                 culture,
```
Comment (jkotas) [inline src/System.Private.CoreLib/src/System/Activator.cs]: I thought that the null check you are about to delete is the one after `It's classic managed type (not WinRT type)`.

Comment (MarcoRossignoli) [reply src/System.Private.CoreLib/src/System/Activator.cs]: Ah...there was a misunderstanding...you copy/paste my comment on `return (o != null) ? new ObjectHandle(o) : null;`. Test case for this coverage found :smile:!
I'll remove this https://github.com/dotnet/coreclr/blob/master/src/System.Private.CoreLib/src/System/Activator.cs#L154!

File: src/System.Private.CoreLib/src/System/Activator.cs (hunk 1 above)
Comment (MarcoRossignoli) [reply src/System.Private.CoreLib/src/System/Activator.cs]: Thank's for tip!

**Comment by jkotas** [comment]
@dotnet-bot test this please

**Pull request merged**